### PR TITLE
Move address into contracts (Compound/Aave/Curve)

### PR DESCRIPTION
**Pull request by leafyoung**

### CODE REVIEW

File: models/credmark/protocols/dexes/curve/curve_finance.py [hunk 1]
```diff
@@ -2,23 +2,79 @@
...
-)
+
+@Model.describe(slug='curve-fi.get-provider',
+                version='1.1',
+                display_name='Curve Finance - Get Provider',
+                description='Get provider contract',
+                input=EmptyInput,
+                output=Contract)
+class CurveFinanceGetProvider(Model):
+    CURVE_PROVIDER_ALL_NETWORK = '0x0000000022D53366457F9d5E68Ec105046FC4383'
```
Comment (newtectonics) [inline models/credmark/protocols/dexes/curve/curve_finance.py:29]: This is cross chain?

File: models/credmark/protocols/dexes/curve/curve_finance.py [hunk 2]
```diff
@@ -2,23 +2,79 @@
...
+                input=EmptyInput,
+                output=Contract)
+class CurveFinanceGetRegistry(Model):
+    def run(self, _) -> Contract:
+        provider = Contract(**self.context.models.curve_fi.get_provider())
+        reg_addr = provider.functions.get_registry().call()
+        return Contract(address=Address(reg_addr).checksum)
+
+
+@Model.describe(slug="curve-fi.get-gauge-controller",
```
Comment (newtectonics) [inline models/credmark/protocols/dexes/curve/curve_finance.py:49]: love it, let's think of a naming convention for this type of thing! the "go get the contract that has a list of contracts" type of thing

**Review by newtectonics — CHANGES_REQUESTED** [review]
looks great! much cleaner

File: models/credmark/protocols/dexes/curve/curve_finance.py (hunk 1 above)
Comment (leafyoung) [reply models/credmark/protocols/dexes/curve/curve_finance.py:29]: It exists where curve made the deployed. From Curve's doc:
> The address provider is deployed to the same address on Ethereum and all sidechains/L2’s where Curve is active.

I checked a few that's correct.
- https://snowtrace.io/address/0x0000000022D53366457F9d5E68Ec105046FC4383
- https://polygonscan.com/address/0x0000000022D53366457F9d5E68Ec105046FC4383

**Pull request merged**